### ISSUE TITLE
Does Sparse prompt version also support no depth input?

### ISSUE THREAD

**Issue by weikaih04** [Robbyant/lingbot-depth#14]
Hi, thanks for the great work! I found that in the code, the sparse prompt version has assert depth is None to force sparse depth input, but in the paper, it claims that when there is no depth input, the model can do monocular depth estimation. Could you guide me on how to achieve this?

**Comment by cherubicXN** [comment]
Hi @weikaih04,

In the paper, we mainly point out that the masked depth modeling (MDM) paradigm is compatible with monocular depth estimation when depth input is not available. In our implementation, this is achieved by setting all depth values to zero, i.e., feeding a dummy depth input.

For a fair comparison on monocular depth estimation, the MDM-pretrained model should be used as an initialization and then further fine-tuned on the MDE task.
